File: Client/PermissionService.cs
using KolibSoft.AuthStore.Core.Filters;
using KolibSoft.AuthStore.Core.Models;
using KolibSoft.Catalogue.Client;

namespace KolibSoft.AuthStore.Client;

public class PermissionService : CatalogueService<PermissionModel, PermissionFilters>
{
    public PermissionService(HttpClient httpClient, string uri) : base(httpClient, uri) { }
}

File: docs/lib/services/permission_service.js
import { PermissionModel } from "../models/permission_model.js";
import { CatalogueService } from "../modules/catalogue.js"

class PermissionService extends CatalogueService {

    /**
     * @param {(json: {}) => PermissionModel} creator 
     * @param {(input: RequestInfo | URL, init?: RequestInit | undefined) => Promise<Response>} fetch 
     * @param {string} uri 
     */
    constructor(creator, fetch, uri) {
        super(creator, fetch, uri);
    }

}

export {
    PermissionService
}